cli/lib/server.js

```diff
@@ -1,126 +1,185 @@
+                console.log(`\tw4 help run\n`)
+                console.log('for more information.')
+                process.exit(1);
+            }
+        } else  {
+            throw e;
         }
     });
 
-    const wsServer = new WebSocketServer({ noServer: true });
-    wsServer.on("connection", socket => {
+    const wSocketServer = new WebSocketServer({ noServer: true });
+    wSocketServer.on("connection", socket => {
         socket.on("message", message => {
             console.log(message);
         });
     });
-    server.on("upgrade", (request, socket, head) => {
-        wsServer.handleUpgrade(request, socket, head, socket => {
-            wsServer.emit("connection", socket, request);
+    webServer.on("upgrade", (request, socket, head) => {
+        wSocketServer.handleUpgrade(request, socket, head, socket => {
+            wSocketServer.emit("connection", socket, request);
         });
     });
 
-    let reloadTimeoutId = 0;
-    fs.watch(path.dirname(cartFile), (event, file) => {
-        if (file == path.basename(cartFile)) {
-            clearTimeout(reloadTimeoutId);
-            reloadTimeoutId = setTimeout(() => {
-                let sentReload = false;
-                for (let client of wsServer.clients) {
-                    client.send(hot ? "hotswap" : "reload");
-                    sentReload = true;
-                }
-                if (sentReload) {
-                    const cart = path.basename(cartFile);
-                    if (hot) {
-                        console.log(`✔ Hot swapped ${cart} (press R for full reload)`);
-                    } else {
-                        console.log(`✔ Reloaded ${cart}`);
-                    }
+    const hot = opts.hot;
+
+    if (shouldWatch) {
+        console.log(opts.settleTime);
+        watch_path(cartFile, opts.settleTime, () => {
+            let sentReload = false;
+            for (let client of wSocketServer.clients) {
+                client.send(hot ? "hotswap" : "reload");
+                sentReload = true;
+            }
+            if (sentReload) {
+                const cart = path.basename(cartFile);
+                let now = new Date();
+                let time_string = `${now.getHours()}:${now.getMinutes()}`;
+                if (hot) {
+                    console.log(`${time_string} 🔄🔥 Hot swapped ${cart} (press R for full reload)`);
+                } else {
+                    console.log(`${time_string} 🔄 Reloaded ${cart}`);
                 }
-            }, 50);
-        }
-    });
+            }
+        });
+    }
 }
 exports.start = start;
 
-async function getStdinBuffer() {
-	const result = [];
-	var length = 0;
+async function wait (ms) {
+    return new Promise((resolve, reject) => setTimeout(resolve, ms));
+}
+
+// Watches a file-path, and calls handleFreshFile when the file is replaced or modified,
+// once all changes have settled down.
+async function watch_path(path, pollTime, handleFreshFile) {
```

Comment:
```suggestion
async function watchPath(path, pollTime, handleFreshFile) {
```

Style nitpicks, could you please use `camelCase` instead of `snakey_case` when working in the JS/TS parts of the codebase?

As an aside, I think it'd be good to beef up our linting rules to make styling a little more uniform, although I'm not a stickler for code formatting.